### PR TITLE
feat(core): drop support for TypeScript 4.2 and 4.3

### PR DESCRIPTION
**Pull request by crisbeto**
Bumps the minimum required TypeScript version to 4.4.2 and removes the integration tests for 4.1, 4.2 and 4.3.

BREAKING CHANGE:
TypeScript versions older than 4.4.2 are no longer supported.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -894,9 +894,11 @@ workflows:
...
             - build-ivy-npm-packages
             - legacy-unit-tests-saucelabs
-      - components-repo-unit-tests:
-          requires:
-            - build-npm-packages
+      # TODO(devversion): re-enable once the Angular components repo has been updated to deal
+      # with partial compilation APF v13 package output (i.e. with the linker processing).
+      #- components-repo-unit-tests:
+      #    requires:
+      #      - build-npm-packages
```
Comment (crisbeto) [inline .circleci/config.yml:901]: For reference, I've talked to Paul about this and the check will be disabled in his APF PR anyway.

**Review by gkalpak — APPROVED** [review]

**Review by JoostK — APPROVED** [review]

**Review by josephperrott — APPROVED** [review]
LGTM

**Review by alan-agius4 — APPROVED** [review]

File: goldens/size-tracking/aio-payloads.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "master": {
       "uncompressed": {
         "runtime": 4454,
-        "main": 463655,
+        "main": 461446,
```
Comment (gkalpak) [inline goldens/size-tracking/aio-payloads.json:6]: :tada:

**Review by petebacondarwin — APPROVED** [review]
Reviewed-for: size-tracking

**Review by alxhub — APPROVED** [review]

File: goldens/size-tracking/aio-payloads.json (hunk 2 above)
Comment (IgorMinar) [reply goldens/size-tracking/aio-payloads.json:6]: awesome

**Review by IgorMinar — APPROVED** [review]
LGTM

**Review by IgorMinar — APPROVED** [review]
LGTM 

Reviewed-for: size-tracking

**Comment by dylhunn** [comment]
@crisbeto FYI, Looks like this needs a presubmit status and some rebasing to merge cleanly.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit c14085e43493f0a1eaea1df949cbf6f3b13b72a0.

**Pull request closed without merging**